exporter/splunkhecexporter/client.go

```diff
@@ -611,9 +619,11 @@ func (c *client) postEvents(ctx context.Context, events io.Reader, headers map[s
 	defer resp.Body.Close()
 
 	err = splunk.HandleHTTPCode(resp)
+	if err != nil {
+		return err
+	}
 
-	io.Copy(ioutil.Discard, resp.Body)
-
+	_, err = io.Copy(ioutil.Discard, resp.Body)
 	return err
```

Comment:
you always need to "discard" everything in Body.

so probably have something like:

```
	err = splunk.HandleHTTPCode(resp)

	_, errCopy = io.Copy(ioutil.Discard, resp.Body)
	return multierr.Combine(err, errCopy)
```